### PR TITLE
Improve custom ops aliasing error message

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134688
* #134692
* #134690
* #134491
* #134490
* #134466

Fixes https://github.com/pytorch/pytorch/issues/134278

Test Plan:
- tested locally

### CODE REVIEW

File: torch/_library/custom_ops.py [hunk 1]
```diff
@@ -332,12 +332,16 @@ def backend_impl(*args, **kwargs):
                                 fn = self._backend_fns[device_type]
                                 module = inspect.getmodule(fn)
                                 raise RuntimeError(
-                                    f"Tensors returned from custom ops (1) must not "
-                                    f"be inputs to the custom op and (2) may not alias "
-                                    f"any inputs or other returns. Please clone the "
-                                    f"the offending output tensors (e.g. output.clone()) "
-                                    f"or refactor your code. "
-                                    f"Offending op: {self._name} (with implementation in {module})"
+                                    f"{self._name} (with implementation in {module}): "
```
Comment (albanD) [inline torch/_library/custom_ops.py:335]: Aren't inplace ops expected to return the input that they modify inplace as-is though to be able to support autograd?

File: torch/_library/custom_ops.py [hunk 2]
```diff
@@ -332,12 +332,16 @@ def backend_impl(*args, **kwargs):
...
+                                    f"The output of this custom operator (1) must not "
```
Comment (albanD) [inline torch/_library/custom_ops.py:336]:
```suggestion
                                    f"The output of a custom operator (1) must not "
```

File: torch/_library/custom_ops.py (hunk 1 above)
Comment (zou3519) [reply torch/_library/custom_ops.py:335]: We don't support that in the compile path or the torch.library.custom_op API

**Review by yushangdi — APPROVED** [review]

**Pull request closed without merging**